### PR TITLE
Feat authentication

### PR DESCRIPTION
**Pull request by nakulsharma15**
Feat: Added Authentication (Login, Signup, Login using Guest Credentials)

### CODE REVIEW

File: src/utils/handleAuth.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ const signupUser = createAsyncThunk("auth/signupUser",
             toast.success("You're successfully logged in.", { style: toastStyle });
             return res.data;
         } catch (err) {
+            console.log(err);
```
Comment (vaibhav-mahalle) [inline src/utils/handleAuth.js:13]: remove console log if not necessary.

**Review by vaibhav-mahalle — APPROVED** [review]
the code looks clean, good formatting no changes required.

File: src/utils/handleAuth.js (hunk 1 above)
Comment (nakulsharma15) [reply src/utils/handleAuth.js:13]: Ok, Thank you for the suggestion

**Review by krishanu2911 — APPROVED** [review]
Well Written code 
Nicely Managed CSS no changes required 
Good to go 👍🏻

**Pull request merged**